code/controllers/subsystem/discord.dm

```diff
@@ -113,3 +113,12 @@ SUBSYSTEM_DEF(discord)
 /datum/controller/subsystem/discord/proc/id_clean(input)
 	var/regex/num_only = regex("\[^0-9\]", "g")
 	return num_only.Replace(input, "")
+
+/datum/controller/subsystem/discord/proc/grant_role(id)
+	// Ignore this shit if config isnt enabled for it
+	if(!CONFIG_GET(flag/enable_discord_autorole))
+		return
+
+	// Make the request
+	var/response = rustg_http_request_blocking(RUSTG_HTTP_METHOD_PUT, "https://discordapp.com/api/guilds/[CONFIG_GET(string/discord_guildid)]/members/[id]/roles/[CONFIG_GET(string/discord_roleid)]", "",  "{\"Authorization\":\"Bot [CONFIG_GET(string/discord_token)]\"}")
+	WRITE_LOG(GLOB.discord_api_log, "DISCORD API: [response]")

```

Comment:
is it a json string?

Reply:
also please log the url as well :D

Reply:
Yes. Do you want each item of it to be newlined?

Reply:
no, just slap it in